### PR TITLE
feat!: Use 32-bit vector_math

### PR DESCRIPTION
**Pull request by renancaraujo**
# Description
Change to 32-bit vector_math to not have to not have incompatible types with 3D and shader code.

## Checklist

- [x] I have followed the [Contributor Guide] when preparing my PR.
- [x] I have updated/added tests for ALL new/updated/fixed functionality.
- [x] I have updated/added relevant documentation in `docs` and added dartdoc comments with `///`.
- [x] I have updated/added relevant examples in `examples` or `docs`.

## Breaking Change?

- [x] Yes, this PR is a breaking change.
- [ ] No, this PR is not a breaking change.

## Related Issues

[Contributor Guide]: https://github.com/flame-engine/flame/blob/main/CONTRIBUTING.md
[Conventional Commit]: https://conventionalcommits.org
[CHANGELOG]: https://github.com/flame-engine/flame/blob/main/CHANGELOG.md

https://github.com/flame-engine/flame/issues/3406

### CODE REVIEW

File: packages/flame/test/extensions/aabb_test.dart [hunk 1]
```diff
@@ -7,7 +7,7 @@ import 'package:test/test.dart';
 void main() {
   group('Aabb2Extension', () {
     // aabb2 is an axis aligned bounding box between a min and a max
-    // see https://api.flutter.dev/flutter/vector_math_64/Aabb2-class.html
+    // see https://api.flutter.dev/flutter/package:vector_math/vector_math.dart/Aabb2-class.html
```
Comment (luanpotter) [inline packages/flame/test/extensions/aabb_test.dart:10]: new link is broken

**Review by spydon — APPROVED** [review]
Lgtm

**Review by spydon — APPROVED** [review]
Lgtm!

**Comment by renancaraujo** [comment]
After changing the imports across the board, i think we should not merge this. 
Specially because some crucial tests on collision detection reports some false negatives after this change. This lack of precisioon can mount up and create noticible differences in value changes in sequence (see how we changed tollerance epsilons on the sequence effect test). 
I know that nobody should care about values after 5/6 digits after the comma speciallly when it comes down to pixels on users screen, but  when it comes down to rotation, scaling and collision detection, i feel very incofrtable with the changes i had to make on tests in order t.

**Pull request closed without merging**